src/scripts/main.js

```diff
@@ -1,3 +1,38 @@
 'use strict';
 
-// write code here
+const bodyElement = document.body;
+
+const dataSalary = [...bodyElement.querySelectorAll('li[data-salary]')];
+const listPeople = [...bodyElement.querySelectorAll('li')];
+
+const salaries = dataSalary
+  .map((elem) => +elem.getAttribute('data-salary').replace(/[$,]+/g, ''))
+  .sort((a, b) => b - a);
+
+const sortedArray = [];
+
+function sortList(list) {
+  for (let i = 0; i < list.length; i++) {
+    const salary = salaries[i];
+
+    const person = list.filter(
+      (elem) =>
+        +elem.getAttribute('data-salary').replace(/[$,]+/g, '') === salary,
+    );
```

Comment:
The filter method returns an array, so 'person' will always be an array, even if it contains only one element. Consider using 'find' instead of 'filter' if you expect only one element to match the condition.